docs/sidecar-design.md

```diff
@@ -0,0 +1,184 @@
+# Datamon sidecar design
+
+This document describes the current design of datamon sidecars.
+
+<!-- Ideas to improve this design may be found [here](proposals/sidecar-design.md) -->
+
+## Design
+
+The current design favors the sidecar container approach, with bespoke signaling between containers,
+over the CSI driver approach. Therefore,  datamon is not available as a kubernetes persistent volume plugin.
+
+This design choice stems from the current inability by Kubernetes CSI API to handle ephemeral volumes.
+Managing many short lived kubernetes volumes is at the moment not practical.
```

Comment:
This is specific to GKE, CSI should have the support for it shortly but is far away from GKE.

Reply:
Indeed you are correct. This part was added some time ago. Now updated to refer to possible future CSI implementation, when k8s v1.16 is available